### ISSUE TITLE
Compilation of flex fails on a number of platforms

### ISSUE THREAD

**Issue by djerius** [plicease/Alien-flex#2]
There are build failures (see [CPAN testers](https://www.cpantesters.org/distro/A/Alien-flex.html?oncpan=1&distmat=1&version=0.12&grade=3) as well as comments below) due to a segv in the `flex` build  (when there isn't a system executable available):

```
./stage1flex   -o stage1scan.c ./scan.l
make[3]: *** [Makefile:1696: stage1scan.c] Segmentation fault
```

I attempted to install `Alien::flex` in a number of the official Perl5 docker images, with results that, when compared with CPAN testers, implies that this is correlated more with OS than Perl version (certainly the underlying compilation should be independent of Perl version).

The Gitlab ci file is [here](https://gitlab.com/djerius/test-p5-alien-flex/-/blob/master/.gitlab-ci.yml); it just loads the Perl5 docker version and attempts to install `Alien::flex`

You can find the results here:

https://gitlab.com/djerius/test-p5-alien-flex/pipelines/119332742

The builds seem to start failing when compiled on Debian buster.

**Comment by djerius** [comment]
I noticed that there is a docker image for a 5.26 build on Debian _stretch_ (the version before _buster_).  `Alien::flex` successfully bulds against that; it fails when built against _buster_.

See:

https://gitlab.com/djerius/test-p5-alien-flex/pipelines/119344694

**Comment by plicease** [comment]
It wouldn't surprise me if the upstream flex doesn't work on modern Linux tbh.  I had to apply patches recently to get m4 to work on the latest versions of both Debian and Fedora because of new glibc being used in those distros.  I got the patch from the Debian package, but it worked with both Debian and Fedora:

https://github.com/Perl5-Alien/Alien-m4/commit/4f02dd9fd0d93216ac702858344a6f1624739ab1

Might be related: https://github.com/westes/flex/issues/436

The issue is a little hard to follow because the reporter has an axe to grind against autoconf, but it might be worth seeing if we get the same fail against HEAD I think there hasn't been a release for a couple of years.

**Comment by plicease** [comment]
Yup it fails consistently with the new version of Debian
https://github.com/plicease/Alien-flex/pull/3
This should be reported upstream, but we can also apply a patch if a new release isn't coming any time soon.

**Comment by djerius** [comment]
See PR #4

**Comment by plicease** [comment]
0.14 is off to CPAN.

**Comment by plicease** [comment]
currently we have a share fail on macOS and Windows, at least in CI.  I wasn't able to reproduce the macOS fail locally (it built fine on my machine).  Seems like flex comes with macOS (or maybe comes with Xcode) so it is urgent.  The fail on Windows is more serious since it is a pain to install flex there.  There are PRs for both of these hopefully can figure them out at some point.
